### PR TITLE
(REF) Allow subclasses of AbstractTokenSubscriber to override the determination of active tokens.

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
This is extracted from PRs #13174 and #12012  By refactoring the determination of active tokens into a separate function `getActiveTokens()`, subclasses of `Civi\Token\AbstractTokenSubscriber` can override this definition if desired.  

No functional change.

### CODE REVIEW

File: Civi/Token/AbstractTokenSubscriber.php [hunk 1]
```diff
@@ -158,6 +156,21 @@ public function evaluateTokens(TokenValueEvent $e) {
...
+   * To handle variable tokens, override this function and return the active tokens.
+   *
+   * @param \Civi\Token\Event\TokenValueEvent $e
+   *
+   * @return mixed
+   */
+  public function getActiveTokens(TokenValueEvent $e) {
+    $messageTokens = $e->getTokenProcessor()->getMessageTokens();
+    if (!isset($messageTokens[$this->entity])) {
+      return NULL;
```
Comment (seamuslee001) [inline Civi/Token/AbstractTokenSubscriber.php]: @aydun just thinking should this return FALSE as it feels a bit more clearer

Comment (aydun) [reply Civi/Token/AbstractTokenSubscriber.php]: @seamuslee001 Ok, I've changed that to FALSE if you think that's clearer.  I was following the pattern of the pre-existing `prefetch()` just below it, but either works.

**Comment by aydun** [comment]
Test failures don't seem to be related to this change - external service unavailable?

**Comment by aydun** [comment]
@eileenmcnaughton Ok to merge?

**Pull request merged**